manager/integration/tests/test_csi_snapshotter.py

```diff
@@ -1188,3 +1188,304 @@ def finalizer():
                               'default')
 
     request.addfinalizer(finalizer)
+
+
+def test_csi_volumesnapshot_backing_image_basic():
+    """
+    Test Create/Delete BackingImage using VolumeSnapshot with a given Volume
+
+    Setup
+    - Create a VolumeSnapshotClass with type `bi`
+        ```
+        kind: VolumeSnapshotClass
+        apiVersion: snapshot.storage.k8s.io/v1
+        metadata:
+            name: longhorn-snapshot-vsc
+        driver: driver.longhorn.io
+        deletionPolicy: Delete
+        parameters:
+            type: bi
+        ```
+
+    Given
+    - A Volume `test-vol` attached to a workload
+
+    When
+    - Creating the VolumeSnapshot
+        ```
+        apiVersion: snapshot.storage.k8s.io/v1beta1
+        kind: VolumeSnapshot
+        metadata:
+            name: test-snapshot-backing
+        spec:
+            volumeSnapshotClassName: longhorn-snapshot-vsc
+            source:
+                persistentVolumeClaimName: test-vol
+        ```
+
+    Then
+    - A BackingImage is created with the following properties
+        ```
+        apiVersion: longhorn.io/v1beta2
+        kind: BackingImage
+        metadata:
+            name: `snapshot-${VolumeSnapshot.uuid}`
+            namespace: longhorn-system
+        spec:
+            sourceType: export-from-volume
+            sourceParameters:
+                volume-name: test-vol
+                export-type: raw
+        ```
+
+    When
+    - Deleting the VolumeSnapshot
+        ```
+        > kubectl delete vs/test-snapshot-backing
+        ```
+
+    Then
+    - The BackingImage is deleted as well
```

Comment:
We should add a check that volume is not deleted.

Reply:
In this testcase
create volumeA -> create snapshot -> create volumeB from VolumeSnapshot -> verify data

When doing deletion, we need to delete VolumeB first so we can delete VolumeSnapshot and lead to BackingImage deletion.
because
- VolumeSnapshot is binding with BackingImage
- if there is any volume using the BackingImage, the BackingImage can't be deleted.